arch/risc-v/src/mpfs/mpfs_irq.c

```diff
@@ -50,6 +50,19 @@ void up_irqinitialize(void)
 
   up_irq_save();
 
+  /* Complete possibly claimed IRQs in PLIC (for current hart) in case
+   * of warm reboot, e.g. after a crash in the middle of IRQ handler.
+   * This has no effect on non-claimed or disabled interrupts.
+   */
+
+  uintptr_t claim_address = mpfs_plic_get_claimbase();
+
+  int irq;
+  for (irq = MPFS_IRQ_EXT_START; irq < NR_IRQS; irq++)
```

Comment:
Not sure why there's int id; for (id = ... below, I'd rather go with the declaration within the loop:

`for (int irq = MPFS_IRQ_EXT_START; irq < NR_IRQS; irq++)`

as irq (or id) is only used within the loop..

Reply:
I'd rather move the variable declarations into beginning of the block (function); K&R / C89 style is more NuttX than C99 ->?

Reply:
Even better